Chore Release 7.4.0

y3rsh opened the pull request:
# This is the working branch for the 7.4.0 release
## This release is for HEPA/UV firmware update

### TODO

~~- [ ] @ahiuchingau  [is there more work to do?](https://github.com/Opentrons/opentrons/pull/15413#pullrequestreview-2123838535)~~
~~- [ ] @ahiuchingau once ready, merge to edge then cherry pick [gripper version PR](https://github.com/Opentrons/opentrons/pull/15413) into this branch `chore_release-7.4.0`~~
- [x] @vegano1 cherry pick [this PR](https://github.com/Opentrons/opentrons/pull/15311) into this branch `chore_release-7.4.0` 
- [x] #15531 merged before build
- [x] @y3rsh create a tag when ready to build for https://github.com/Opentrons/ot3-firmware/pull/782

vegano1 committed:
feat(hardware): added safety relay active state to HepaUVState and safety_relay_inactive error (#15311)

vegano1 commented:
@y3rsh 
PR has been cherry picked
https://github.com/Opentrons/opentrons/pull/15311

ecormany committed:
chore: 7.4.0 release notes (#15531)

# Overview

Indicate that 7.4.0 adds support for HEPA/UV.

# Test Plan

Check in-app after alpha 0 tag is cut.

# Changelog

app-shell and api notes.

# Risk assessment

nil

shlokamin committed:
fix(api): Parse all RTP fields strictly to fix flakiness (#15187) (#15536)

closes https://opentrons.atlassian.net/browse/RESC-287

---------

Co-authored-by: Max Marrone <max@opentrons.com>
Co-authored-by: Edward Cormany <edward.cormany@opentrons.com>
Co-authored-by: Sanniti Pimpley <sanni-t@users.noreply.github.com>

shlokamin committed:
ci(app-shell): update macos github runner version (#15559)

y3rsh committed:
RQA-2830 fix mac dmg to be x64 binary and use x64 bundled python to pip install (#15624)

## Replaces #15618 with PR against branch name ending in
`app-build-internal` to trigger a build

## Overview

- <https://opentrons.atlassian.net/browse/RQA-2830>
- `macos-11` GitHub runner was intel and now we use macos-latest that is
arm
[doc](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories)
- specify --x64 argument to electron-builder
- ~~per
[doc](https://github.com/actions/setup-python?tab=readme-ov-file#supported-architectures)
x64 python should be installed so the pip install will be the right
platform. I'm not confident in this.~~
- see comment in #15618
- we need to use the python we download on mac to do the pip install so
the installed packages match the platform correctly

## Test Plan

- [x] build locally and see that it works
- [x] watch CI and install the built package
- [x] validate that once the package dmg is x64 that analysis works
(python has packages with the right arch)

# Review requests

- This the right approach?

# Risk assessment

High because our .dmg currently cannot analyze protocols 2.0.0-alpha.0
internal release build.

y3rsh committed:
build: Sign windows builds via remote key storage (#15718) (#15735)

## `cherry-pick` 246efcbb52

Must update `chore_release-7.4.0` so that windows builds work and we may
test an update. To understand the changes see #15718

Co-authored-by: Seth Foster <seth@opentrons.com>

y3rsh committed:
fix(scripts): setuptools v71 removed the need for extern (#15717) (#15741)

## `cherry-pick` afaa7177fe

Must update `chore_release-7.4.0` so that Linux app builds work due to a
change in setuptools. To understand the changes see #15717

Co-authored-by: Ryan Howard <ryan.howard@opentrons.com>

nusrat813 reviewed (approved):
lgtm

vegano1 reviewed (approved):
Oh yeah!
